core/server/version.go

```diff
@@ -15,10 +21,66 @@ var (
 )
 
 func (cs *coreServer) GetVersion(ctx context.Context, msg *pb.GetVersionRequest) (*pb.GetVersionResponse, error) {
+	u := &unstructured.Unstructured{}
+
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "Namespace",
+	})
+
+	var ns string
+
+	if msg.Namespace != "" {
+		ns = msg.Namespace
+	} else {
+		ns = wego.DefaultNamespace
+	}
+
+	key := client.ObjectKey{
+		Name: ns,
+	}
```

Comment:
So, you're right that this can be pretty much any namespace - but I'm not sure the frontend knows what namespace to use :thinking: 

What if instead of `Get`ing a specific namespace, we `List` filtered by `{"app.kuberenetes.io/part-of": "flux"}` and just grab the first result we find? After a bit of grepping - I think this helper does the filtering? https://github.com/weaveworks/weave-gitops/blob/main/core/server/server.go#L86

Reply:
OK, can do it.

From frontend we don't provide any namespace. This param was added just for passing the test (because when running the test it could not find the "flux-system" namespace). If the namespace is not provided (like when we make the request via the API, `{}` with no args), we just default to "flux-system" (and while the app is running normally, the "flux-system" namespace is found).

I need to see if the list helper will return the flux-system/flux-whatever namespace while running the test.